src/drivers/px4fmu/fmu.cpp

```diff
@@ -811,67 +844,185 @@ PX4FMU::cycle()
+			// This triggers the port re-configuration
+			rc_scan_begin = 0;
+			// Scan the next protocol
+			rc_scan_state = RC_SCAN_PPM;
+		}
 
-#endif
+		break;
 
-#ifdef HRT_PPM_CHANNEL
+	case RC_SCAN_ST24:
+		if (rc_scan_begin == 0) {
+			rc_scan_begin = now;
+//			// Configure serial port for DSM
+			dsm_config(_rcs_fd);
+//			/* for PixRacer R07, this signal is active low */
+//			/* for PixRacer R12, this signal is active high */
+			stm32_gpiowrite(GPIO_SBUS_INV, 1);
+
+		} else if (now - rc_scan_last_lock < rc_scan_max
+			   || now - rc_scan_begin < rc_scan_max) {
+
+			if (newBytes > 0) {
+				// parse new data
+				uint8_t st24_rssi, rx_count;
+
+				rc_updated = false;
+
+				for (unsigned i = 0; i < newBytes; i++) {
+					/* set updated flag if one complete packet was parsed */
+					st24_rssi = RC_INPUT_RSSI_MAX;
+					rc_updated |= (OK == st24_decode(_rcs_buf[i], &st24_rssi, &rx_count,
+									 &raw_rc_count, raw_rc_values, input_rc_s::RC_INPUT_MAX_CHANNELS));
+				}
 
-	// see if we have new PPM input data
-	if ((ppm_last_valid_decode != _rc_in.timestamp_last_signal) &&
-	    ppm_decoded_channels > 3) {
-		// we have a new PPM frame. Publish it.
-		_rc_in.channel_count = ppm_decoded_channels;
+				if (rc_updated) {
+					// we have a new ST24 frame. Publish it.
+					fill_rc_in(raw_rc_count, raw_rc_values, now,
+						   false, false, dsm_frame_drops);
+					rc_scan_last_lock = now;
+					rc_scan_locked = true;
+				}
+			}
 
-		if (_rc_in.channel_count > input_rc_s::RC_INPUT_MAX_CHANNELS) {
-			_rc_in.channel_count = input_rc_s::RC_INPUT_MAX_CHANNELS;
+		} else if (!rc_scan_locked) {
+			// This triggers the port re-configuration
+			rc_scan_begin = 0;
+			// Scan the next protocol
+			rc_scan_state = RC_SCAN_PPM;
 		}
 
-		for (uint8_t i = 0; i < _rc_in.channel_count; i++) {
-			_rc_in.values[i] = ppm_buffer[i];
-		}
+		break;
 
-		_rc_in.timestamp_publication = ppm_last_valid_decode;
-		_rc_in.timestamp_last_signal = ppm_last_valid_decode;
+	case RC_SCAN_SUMD:
+		break;
 
-		_rc_in.rc_ppm_frame_length = ppm_frame_length;
-		_rc_in.rssi = RC_INPUT_RSSI_MAX;
-		_rc_in.rc_failsafe = false;
-		_rc_in.rc_lost = false;
-		_rc_in.rc_lost_frame_count = 0;
-		_rc_in.rc_total_frame_count = 0;
+	case RC_SCAN_PPM:
+		if (rc_scan_begin == 0) {
+			rc_scan_begin = now;
+			// Configure timer input pin for CPPM
+			stm32_configgpio(GPIO_PPM_IN);
+
+		} else if (now - rc_scan_last_lock < rc_scan_max
+			   || now - rc_scan_begin < rc_scan_max) {
+
+			// see if we have new PPM input data
+			if ((ppm_last_valid_decode != _rc_in.timestamp_last_signal)
+			    && ppm_decoded_channels > 3) {
+				// we have a new PPM frame. Publish it.
+				rc_updated = true;
+				fill_rc_in(ppm_decoded_channels, ppm_buffer, now,
+					   false, false, 0);
+				rc_scan_last_lock = now;
+				rc_scan_locked = true;
+			}
+
+		} else if (!rc_scan_locked) {
+			// This triggers the port re-configuration
+			rc_scan_begin = 0;
+			// Scan the next protocol
```

Comment:
This would be the location to de-configure the PPM in pin: Configure it as GPIO input without pullup or pulldown and do NOT set the PPM alternate function.